packages/block-library/src/social-link/edit.js

```diff
@@ -99,14 +113,14 @@ const SocialLinkEdit = ( {
 		iconBackgroundColor,
 		iconBackgroundColorValue,
 	} = context;
-	const [ showURLPopover, setPopover ] = useState( false );
 	const classes = classNames( 'wp-social-link', 'wp-social-link-' + service, {
 		'wp-social-link__is-incomplete': ! url,
 		[ `has-${ iconColor }-color` ]: iconColor,
 		[ `has-${ iconBackgroundColor }-background-color` ]:
 			iconBackgroundColor,
 	} );
 
+	const [ showPopover, setShowPopover ] = useState( ! url && ! service );
```

Comment:
The inital state change allows Popover to be opened for freshly inserted blocks.